### PR TITLE
Docs: new light theme and various other updates

### PR DESCRIPTION
**Pull request by mosra**
## Motivation and Context

What's done:

- The m.css submodule is updated with mostly CSS-related changes in order to allow light themes work better, plus a bundle of changes and fixes done when porting habitat-api python docs to it
- Cross-linking to the `quaternion` types as well (originally it was using some broken `*.inv` file and nothing worked)
- New light theme using Habitat colors (there's still quite some work left to be done by some designer, you'll see once you start using the more colorful features of m.css)
- Using Pygments' builtin "pastie" theme for code highlighting to fit better with the light theme (feel free to adapt / change that tho)
- The Magnum Python Bindings have a stable URL scheme now, which means existing links didn't work anymore. The `*.inv` file got updated to fix that.
- The `build_docs.sh` helper got moved into `docs/` and it now also does CSS compilation and has a possibility to use an external m.css project instead of the submodule (I'm using that a lot for iterations)
- The docs now have a favicon, brand logo and a theme color so e.g. your android browser will match its chrome color with the page; also there's a copyright footer added

## How Has This Been Tested

```sh
git submodule update --init
cd docs
./build.sh
open ../build/docs/habitat-sim/index
```

### CODE REVIEW

**Comment by mosra** [comment]
Done in 810e30304b193af528d8f6d45d5729bc6e52f637, thanks. Also did a minor change on the API side (facebookresearch/habitat-api@e66bfa850e272443aaa5093423fc22412e45e831), felt more consistent that way.

File: docs/pages/index.rst [hunk 1]
```diff
@@ -5,8 +5,18 @@ A flexible, high-performance 3D simulator with configurable agents, multiple
...
+over 10,000 FPS multi-process on a single GPU!
+
+`Tutorials`_
+============
+
+.. TODO: this is waiting on m.css to propagate page titles to links
+
+-   `New Actions <std:doc:new-actions>`
+-   `Notebooks <std:doc:notebooks>`
+-   `Creating a stereo agent <std:doc:stereo-agent>`
```
Comment (erikwijmans) [inline docs/pages/index.rst]: Should this be links yet?

Comment (mosra) [reply docs/pages/index.rst]: If you mean the TODO, no, that still holds ... i had to prioritize to not spend next 16 years on the generator :)

Comment (erikwijmans) [reply docs/pages/index.rst]: Yep, no problem, just wanted to make sure there wasn't something I was missing :)

**Pull request merged**